### PR TITLE
OEL-94: Finalize ingestion

### PR DESCRIPTION
**Pull request by claudiu-cristea**

### CODE REVIEW

File: src/Api/FileIngestionApi.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * @inheritDoc
+     */
+    public function getRequestMultipartStreamElements(): array
+    {
+        $parts = parent::getRequestMultipartStreamElements();
+
+        if ($file = $this->getFile()) {
+            $parts['file'] = $this->jsonEncoder->encode($file, 'json');
```
Comment (drishu) [inline src/Api/FileIngestionApi.php]: If I set a binary string it complains about Malformed UTF-8 characters, I think we don't need to encode at all.

Also in ApiBase::getRequestBody, we always set `'Content-Type' => 'application/json'`, might be we need to adapt, only actual tests will tell.

File: src/Api/FileIngestionApi.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\EuropaSearchClient\Api;
+
+use OpenEuropa\EuropaSearchClient\Contract\FileIngestionApiInterface;
+use OpenEuropa\EuropaSearchClient\Model\Ingestion;
```
Comment (drishu) [inline src/Api/FileIngestionApi.php]: unused, same for TextIngestionApi

File: src/Api/IngestionApiBase.php [hunk 3]
```diff
@@ -49,12 +84,20 @@ public function getConfigSchema(): array
...
             'database' => $this->getConfigValue('database'),
             'uri' => $this->getUri(),
-            'reference' => $this->getReference(),
         ] + parent::getRequestUriQuery($uri);
+
+        if ($languages = $this->getLanguages()) {
+            $query['language'] = $this->jsonEncoder->encode($languages, 'json');
+        }
+        if ($reference = $this->getReference()) {
+            $query['reference'] = $this->jsonEncoder->encode($reference, 'json');
```
Comment (drishu) [inline src/Api/IngestionApiBase.php]: if the purpose is to encode we should use:
```suggestion
            $query['reference'] = rawurlencode($reference);
```
with json encode we will get double quoted strings like ""ref1""

File: tests/src/Api/FileIngestionApiTest.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+     * @dataProvider providerTestFileIngestion
+     *
+     * @param array $clientConfig
+     * @param array $responses
+     * @param mixed $expectedResult
+     */
+    public function testFileIngestion(array $clientConfig, array $responses, $expectedResult): void
+    {
+        $actualResult = $this->getTestingClient($clientConfig, $responses)
+            ->ingestFile('http://example.com', chr(0) . 'saDD*&6778)(&88');
```
Comment (drishu) [inline tests/src/Api/FileIngestionApiTest.php]: Would it be worth here passing the binary of a real file (adding one in a fixtures folders and then file_get_contents..) ?

File: tests/src/ClientTest.php [hunk 5]
```diff
@@ -57,6 +58,7 @@ public function testContainer(): void
         $this->assertInstanceOf(FacetApiInterface::class, $container->get('facet'));
         $this->assertInstanceOf(TokenApiInterface::class, $container->get('token'));
         $this->assertInstanceOf(TokenAwareInterface::class, $container->get('textIngestion'));
+        $this->assertInstanceOf(TokenAwareInterface::class, $container->get('fileIngestion'));
```
Comment (drishu) [inline tests/src/ClientTest.php:66]: Lets add one more line to test FileIngestionApiInterface
```
        $this->assertInstanceOf(FileIngestionApiInterface::class, $container->get('fileIngestion'));
```
same for text and delete

**Review by drishu — CHANGES_REQUESTED** [review]
Let's discuss

File: src/Api/IngestionApiBase.php (hunk 3 above)
Comment (claudiu-cristea) [reply src/Api/IngestionApiBase.php]: We need to pass the raw reference. It's already encoded in `ApiBase::getRequestUri()`.

**Review by drishu — APPROVED** [review]

**Pull request merged**